cmd/babble/commands/root.go

```diff
@@ -0,0 +1,98 @@
+package commands
+
+import (
+	"path/filepath"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+var (
+	config = NewDefaultCLIConfig()
+	logger = logrus.New()
+)
+
+//RootCmd is the root command for Babble
+var RootCmd = &cobra.Command{
+	Use:              "babble",
+	Short:            "babble consensus",
+	TraverseChildren: true,
+	PersistentPreRunE: func(cmd *cobra.Command, args []string) (err error) {
+		if cmd.Name() == VersionCmd.Name() {
+			return nil
+		}
+
+		if err := bindFlagsLoadViper(cmd); err != nil {
+			return err
+		}
+
+		config, err = parseConfig()
+		if err != nil {
+			return err
+		}
+
+		logger.Level = logLevel(config.Babble.LogLevel)
+
+		logger.WithFields(logrus.Fields{
+			"datadir": config.Babble.DataDir,
+			"log":     config.Babble.LogLevel,
+		}).Debug("Config")
+
+		return nil
+	},
+}
+
+//------------------------------------------------------------------------------
+
+//Bind all flags and read the config into viper
+func bindFlagsLoadViper(cmd *cobra.Command) error {
+	// cmd.Flags() includes flags from this command and all persistent flags from the parent
+	if err := viper.BindPFlags(cmd.Flags()); err != nil {
+		return err
+	}
+
+	viper.SetConfigName("config")                                       // name of config file (without extension)
```

Comment:
I think the config name should have the name of the project (babble.[json | yml | ...]) because config.* is not explicit enough (when sharing or storing this file for exemple) maybe something like `babble-config.*` or `babble.config.*` ?

Reply:
good point